google/cloud/spanner/samples/samples.cc

```diff
@@ -1537,6 +1537,35 @@ void ListDatabaseOperationsCommand(std::vector<std::string> argv) {
   ListDatabaseOperations(std::move(client), argv[0], argv[1]);
 }
 
+//! [update-database] [START spanner_update_database]
+void UpdateDatabase(google::cloud::spanner_admin::DatabaseAdminClient client,
+                    std::string const& project_id,
+                    std::string const& instance_id,
+                    std::string const& database_id, bool drop_protection) {
+  google::cloud::spanner::Database db(project_id, instance_id, database_id);
+  google::spanner::admin::database::v1::Database database;
+  database.set_name(db.FullName());
+  database.set_enable_drop_protection(drop_protection);
+  google::protobuf::FieldMask update_mask;
+  update_mask.add_paths("enable_drop_protection");
+  auto updated = client.UpdateDatabase(database, update_mask).get();
+  if (!updated) throw std::move(updated).status();
+  std::cout << "Database " << updated->name() << " successfully updated.\n";
+}
+//! [update-database] [END spanner_update_database]
+
+void UpdateDatabaseCommand(std::vector<std::string> argv) {
+  if (argv.size() != 4) {
+    throw std::runtime_error(
+        "update-database <project-id> <instance-id> <database-id>"
+        " <drop-protection>");
+  }
+  google::cloud::spanner_admin::DatabaseAdminClient client(
+      google::cloud::spanner_admin::MakeDatabaseAdminConnection());
+  bool drop_protection = (argv[3][0] == 'T');
```

Comment:
assuming `! argv[3].empty()` ?

Reply:
`s[i]` is actually [and weirdly] defined to return (a reference to) a `char()` value when `i == s.size()`.

Reply:
TIL